spec/models/accounting/quickbooks/data_extractor_spec.rb

```diff
@@ -158,5 +158,334 @@ def expect_line_item_amounts(amounts)
+                  'class_ref' => { 'value' => '5000000000000026437', 'name' => loan.id, 'type' => nil },
+                  'department_ref' => nil } }],
+            'id' => '167',
+            'sync_token' => 0,
+            'meta_data' => {
+              'create_time' => '2017-04-18T10:14:30.000-07:00',
+              'last_updated_time' => '2017-04-18T10:14:30.000-07:00' },
+            'txn_date' => '2017-04-18',
+            'total' => '12.30',
+            'private_note' => 'Random stuff' }
+        end
+
+        it do
+          expect(txn.loan_transaction_type_value).to eq('other')
+          expect(txn.managed).to be false
+        end
+
+        describe 'interest with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'disbursement with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'repayment with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'random with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+      end
+
+      describe 'too few line items' do
+        let(:quickbooks_data) do
+          { 'line_items' =>
+            [{ 'id' => '0',
+              'description' => 'Eba',
+              'amount' => '10.99',
+              'detail_type' => 'JournalEntryLineDetail',
+              'journal_entry_line_detail' => {
+                'posting_type' => 'Credit',
+                'entity' => {
+                  'type' => 'Customer',
+                  'entity_ref' => { 'value' => '1', 'name' => "Amy's Bird Sanctuary", 'type' => nil } },
+                'account_ref' => { 'value' => txn_acct.qb_id, 'name' => txn_acct.name, 'type' => nil },
+                'class_ref' => { 'value' => '5000000000000026437', 'name' => loan.id, 'type' => nil },
+                'department_ref' => nil } },
+            ],
+            'id' => '167',
+            'sync_token' => 0,
+            'meta_data' => {
+              'create_time' => '2017-04-18T10:14:30.000-07:00',
+              'last_updated_time' => '2017-04-18T10:14:30.000-07:00' },
+            'txn_date' => '2017-04-18',
+            'total' => '12.30',
+            'private_note' => 'Random stuff' }
+        end
+
+        it do
+          expect(txn.loan_transaction_type_value).to eq('other')
+          expect(txn.managed).to be false
+        end
+      end
+    end
```

Comment:
These tests are good. At some point we might want to refactor this to have a `quickbooks_data_generator` method that takes arguments like `line_items: [{type: :credit, account: txn_acct, amount: 1.23}]`. That would clean this up a lot. But it doesn't have to be done now.

Reply:
I would like to pair on this at that time